python/lsst/ip/diffim/zogy.py

```diff
@@ -231,23 +230,50 @@ def selectPsf(psf, exposure):
         self.im2_psf = selectPsf(psf2, self.science)
 
         # Make sure PSFs are the same size. Messy, but should work for all cases.
-        pShape1 = self.im1_psf.shape
-        pShape2 = self.im2_psf.shape
+        psf1 = self.im1_psf
+        psf2 = self.im2_psf
+        pShape1 = psf1.shape
+        pShape2 = psf2.shape
         if (pShape1[0] < pShape2[0]):
-            self.im1_psf = np.pad(self.im1_psf, ((0, pShape2[0] - pShape1[0]), (0, 0)),
-                                  mode='constant', constant_values=0.)
+            psf1 = np.pad(psf1, ((0, pShape2[0] - pShape1[0]), (0, 0)), mode='constant', constant_values=0.)
         elif (pShape2[0] < pShape1[0]):
-            self.im2_psf = np.pad(self.im2_psf, ((0, pShape1[0] - pShape2[0]), (0, 0)),
-                                  mode='constant', constant_values=0.)
+            psf2 = np.pad(psf2, ((0, pShape1[0] - pShape2[0]), (0, 0)), mode='constant', constant_values=0.)
         if (pShape1[1] < pShape2[1]):
-            self.im1_psf = np.pad(self.im1_psf, ((0, 0), (0, pShape2[1] - pShape1[1])),
-                                  mode='constant', constant_values=0.)
+            psf1 = np.pad(psf1, ((0, 0), (0, pShape2[1] - pShape1[1])), mode='constant', constant_values=0.)
         elif (pShape2[1] < pShape1[1]):
-            self.im2_psf = np.pad(self.im2_psf, ((0, 0), (0, pShape1[1] - pShape2[1])),
-                                  mode='constant', constant_values=0.)
+            psf2 = np.pad(psf2, ((0, 0), (0, pShape1[1] - pShape2[1])), mode='constant', constant_values=0.)
+
+        # PSFs' centers may be offset relative to each other; now fix that!
+        maxLoc1 = np.unravel_index(np.argmax(psf1), psf1.shape)
+        maxLoc2 = np.unravel_index(np.argmax(psf2), psf2.shape)
+        # *Very* rarely happens but if they're off by >1 pixel, do it more than once.
+        while (maxLoc1[0] != maxLoc2[0]) or (maxLoc1[1] != maxLoc2[1]):
+            if maxLoc1[0] > maxLoc2[0]:
+                psf2[1:, :] = psf2[:-1, :]
+            elif maxLoc1[0] < maxLoc2[0]:
+                psf1[1:, :] = psf1[:-1, :]
+            if maxLoc1[1] > maxLoc2[1]:
+                psf2[:, 1:] = psf2[:, :-1]
+            elif maxLoc1[1] < maxLoc2[1]:
+                psf1[:, 1:] = psf1[:, :-1]
+            maxLoc1 = np.unravel_index(np.argmax(psf1), psf1.shape)
+            maxLoc2 = np.unravel_index(np.argmax(psf2), psf2.shape)
+
+        # Make sure there are no div-by-zeros
+        psf1[psf1 < 1e-4] = 1e-4
```

Comment:
This is different than the `1e-3` you used in other code. Using a named constant would help, and it may be worth getting the constant that you use here from the value you set in the C++ code.

Reply:
You're right, there's no logical reason for them to be different so I'll set them all to 1.0e-4. But there's not technically a reason for why they should be the same (the algorithms are different, they just suffer from related issues if the PSF is noisy).
Also I'm confused - there is no C++ code here, so can I set a global python variable and use it in both places?

Reply:
Sorry, Python, not C++. Yes you could set one global python variable and use it in both places -- that makes sense if you can come up with a name that seems usable in the global API. If not, then a constant per module that is not exported would also be fine.